cmd/kubectl-gadget/utils/trace.go

```diff
@@ -200,18 +188,14 @@ func createTraces(trace *gadgetv1alpha1.Trace) error {
 			trace.Spec.Node = node.Name
 		}
 
-		err = traceRestClient.
-			Post().
-			Namespace(trace.ObjectMeta.Namespace).
-			Resource("traces").
-			Body(trace).
-			Do(context.TODO()).
-			Error()
+		newtrace, err := traceClient.GadgetV1alpha1().Traces("gadget").Create(
+			context.TODO(), trace, metav1.CreateOptions{},
+		)
 		if err != nil {
-			traceID, present := trace.ObjectMeta.Labels[GLOBAL_TRACE_ID]
+			traceID, present := newtrace.ObjectMeta.Labels[GLOBAL_TRACE_ID]
```

Comment:
If `err != nil`, I don't think we can rely on `newtrace` being a valid pointer. I think we should still use `trace` here.

Reply:
Good catch!